### PR TITLE
fix focus peaking

### PR DESCRIPTION
**Pull request by ralfbrown**
Current master shows random border pixels (mostly yellow and usually on right/bottom) and random yellow blobs for focus peaking on very small thumbnails (toggle on and off quickly to see them).  This PR fixes those, should speed up the code by eliminating four parallelized loops, and hopefully eliminates some vectorization warnings from LLVM/Xcode.

Also did a bit of code cleanup.

### CODE REVIEW

**Comment by johnny-bit** [comment]
Regarding CI failure - for some reason it looks like "skiptest" on mac now hates xcf.c :/

File: src/common/focus_peaking.h [hunk 1]
```diff
@@ -175,105 +180,42 @@ schedule(static) collapse(2) aligned(focus_peaking, luma_ds:64) reduction(+:sigm
...
+#pragma omp parallel for default(none) \
+  dt_omp_firstprivate(focus_peaking, luma_ds, buf_height, buf_width, six_sigma, four_sigma, two_sigma) \
+  schedule(static) collapse(2)
 #endif
-  for(size_t i = 2; i < buf_height - 2; ++i)
-    for(size_t j = 2; j < buf_width - 2; ++j)
+  for(size_t i = 0; i < buf_height; ++i)
+    for(size_t j = 0; j < buf_width; ++j)
     {
+      static const uint8_t yellow[4] = { 0/*B*/, 255/*G*/, 255/*R*/, 255/*alpha*/ };
```
Comment (TurboGit) [inline src/common/focus_peaking.h:190]: would it be better to have those static const arrays outside of the loops?

Comment (ralfbrown) [reply src/common/focus_peaking.h:190]: `static const` variables are stored in the global initialized-data segment, so the only difference from varying placement is the scope of visibility.  Inside the loop, the optimizer is more likely to see that the values could be replaced by a literal on an instruction (much like a local float[4] often gets optimized away to a vector register).

**Review by TurboGit — APPROVED** [review]
Works for me, thanks !

**Pull request merged**